pwa-shell/src/ui/content-detail/ContentDetail.js

```diff
@@ -4,20 +4,51 @@ import PropTypes from 'prop-types';
 import { Container } from 'reactstrap';
 import ProtectedContentLoginContainer from 'ui/login/ProtectedContentLoginContainer';
 import ContentDetailTopBarContainer from 'ui/content-detail/ContentDetailTopBarContainer';
+import SwipeContentNavigator from 'ui/common/SwipeContentNavigator';
 import PageContainer from 'ui/common/PageContainer';
 import ItemCategoryListContainer from 'ui/common/ItemCategoryListContainer';
 
 class ContentDetail extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      nextURL: '',
+      previousURL: '',
+    };
+  }
   componentDidMount() {
-    this.props.fetchContentDetail();
+    const { location, match } = this.props;
+    this.props.fetchContentDetail(location, match.params);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.match.params !== nextProps.match.params) {
+      const { location, match } = nextProps;
+      this.props.fetchContentDetail(location, match.params);
+    }
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.isUserLogged !== prevProps.isUserLogged) {
-      this.props.fetchContentDetail();
+    if (this.props.nextContent !== prevProps.nextContent || this.props.prevContent !== prevProps.prevContent) {
+      this.checkContentSiblings();
     }
   }
 
+  checkContentSiblings() {
+    const {
+      nextContent,
+      prevContent,
+      contentType,
+    } = this.props;
+
+    const hasNext = Object.keys(nextContent).length > 0;
+    const hasPrev = Object.keys(prevContent).length > 0;
+    this.setState({
+      nextURL: hasNext ? `/content/${contentType}/${nextContent.id}${nextContent.requiresAuth ? '?requiresAuth=true' : ''}` : '',
```

Comment:
you cannot be sure requiresAuth is true for next and prev items; you should check a `requiresAuthMap`

Reply:
Do you mean we add the `requiresAuth` flag of next/previous content to the `requiresAuthMap` before the user actually views the said contents? By what I saw in the code, it will be added to the map (and check) when you actually begin to view the content.

Reply:
sorry, your're right! :)